lib/IRGen/IRGen.cpp

```diff
@@ -490,10 +490,13 @@ static createTargetMachine(IRGenOptions &Opts, ASTContext &Ctx) {
   }
 
   // Create a target machine.
-  llvm::TargetMachine *TargetMachine
-    = Target->createTargetMachine(Triple.str(), CPU,
-                                  targetFeatures, TargetOpts, Reloc::PIC_,
-                                  CodeModel::Default, OptLevel);
+  auto cmodel = CodeModel::Default;
+  if (Triple.isWindowsCygwinEnvironment())
+    cmodel = CodeModel::Large;
```

Comment:
Now I can show you the sample codes.

This is a line of LLVM of Hello.swift.

`
@_TZvOs11CommandLine5_argcVs5Int32 = external global %Vs5Int32, align 4
`

That is `static Swift.CommandLine._argc : Swift.Int32`,
and is a part of the standard library and declared in data section of `cygswiftCore.dll`,
so the application code should read/write the data section of the dll. 

The generated assembly code is as follows.

In default model,
`movl    %edi, _TZvOs11CommandLine5_argcVs5Int32(%rip)`,

and in large model,
`movabsq $_TZvOs11CommandLine5_argcVs5Int32, %rax`
`movl    %ecx, (%rax)`.

In case of 64bit default memory model, since the DLL is located at higher memory (for example, cygswiftcore.dll is loaded at 0x459F40000 > maximum 32bit), and the generated code only references 32bit address, the executable file didn't work correctly.

If we link the static library instead of dll, it might be work correctly, so I think it is hard to say llvm has a bug.